Examples/Io/EDM4hep/src/EDM4hepUtil.cpp

```diff
@@ -0,0 +1,305 @@
+      (float)(momentum4Before[Acts::eMom1] / Acts::UnitConstants::GeV),
+      (float)(momentum4Before[Acts::eMom2] / Acts::UnitConstants::GeV),
+  });
+
+  to.setEDep(-delta4[Acts::eEnergy] / Acts::UnitConstants::GeV);
+}
+
+Measurement EDM4hepUtil::readMeasurement(
+    edm4hep::TrackerHitPlane from,
+    const edm4hep::TrackerHitCollection* fromClusters, Cluster* toCluster,
+    MapGeometryIdFrom geometryMapper) {
+  // no need for digitization as we only want to identify the sensor
+  Acts::GeometryIdentifier geometryId = geometryMapper(from.getCellID());
+
+  // TODO what about the hit index?
+  IndexSourceLink sourceLink{geometryId, 0};
+
+  auto pos = from.getPosition();
+  auto cov = from.getCovMatrix();
+
+  DigitizedParameters dParameters;
+
+  dParameters.indices.push_back(Acts::eBoundLoc0);
+  dParameters.values.push_back(pos.x);
+  dParameters.variances.push_back(cov[0]);
+
+  // TODO cut this out for 1D
+  dParameters.indices.push_back(Acts::eBoundLoc1);
+  dParameters.values.push_back(pos.y);
+  dParameters.variances.push_back(cov[2]);
+
+  dParameters.indices.push_back(Acts::eBoundTime);
+  dParameters.values.push_back(pos.z);
+  dParameters.variances.push_back(cov[5]);
+
+  auto to = createMeasurement(dParameters, sourceLink);
+
+  if (fromClusters != nullptr) {
+    for (auto objectId : from.getRawHits()) {
+      const auto& c = fromClusters->at(objectId.index);
+
+      // TODO get EDM4hep fixed
+      // misusing some fields to store ACTS specific information
+      // don't ask ...
+      ActsFatras::Channelizer::Bin2D bin{(unsigned int)c.getType(),
+                                         (unsigned int)c.getQuality()};
+      ActsFatras::Channelizer::Segment2D path2D;
+      double activation = c.getTime();
+      ActsFatras::Channelizer::ChannelSegment cell{bin, path2D, activation};
+
+      toCluster->channels.push_back(cell);
+    }
+  }
+
+  return to;
+}
+
+void EDM4hepUtil::writeMeasurement(const Measurement& from,
+                                   edm4hep::MutableTrackerHitPlane to,
+                                   const Cluster* fromCluster,
+                                   edm4hep::TrackerHitCollection& toClusters,
+                                   MapGeometryIdTo geometryMapper) {
+  std::visit(
+      [&](const auto& m) {
+        Acts::GeometryIdentifier geoId = m.sourceLink().geometryId();
+
+        if (geometryMapper != nullptr) {
+          // no need for digitization as we only want to identify the sensor
+          to.setCellID(geometryMapper(geoId));
+        }
+
+        auto parameters = (m.expander() * m.parameters()).eval();
+
+        to.setTime(parameters[Acts::eBoundTime] / Acts::UnitConstants::ns);
+
+        to.setType(EDM4hepUtil::EDM4HEP_ACTS_POSITION_TYPE);
+        // TODO set uv (which are in global spherical coordinates with r=1)
+        to.setPosition({parameters[Acts::eBoundLoc0],
+                        parameters[Acts::eBoundLoc1],
```

Comment:
This will just be zero in case of lower dimension (i.e. 1D) measurements, right?